sdc_client/client.py

```diff
@@ -311,18 +327,16 @@ def get_jmx(streamsets: IStreamSets, query: str) -> dict:
     return _get_client(streamsets).get_jmx(query)
 
 
-def get_jmxes_async(queries: List[Tuple[IStreamSets, str]]) -> List[Dict]:
-    async def execute_requests(queries: List[Tuple[IStreamSets, str]]):
-        tasks = [asyncio.create_task(
-            _AsyncStreamSetsApiClient(streamset_).get_jmx(query)
-        ) for streamset_, query in queries]
-        await asyncio.wait(tasks)
-        return [task.result() for task in tasks]
-    loop = asyncio.new_event_loop()
-    jmxes = loop.run_until_complete(execute_requests(queries))
-    loop.close()
-    return jmxes
-
+def get_jmxes_async(queries: List[Tuple[IStreamSets, str]], return_exceptions=False) -> List[Dict]:
+    async def execute_requests(queries_: List[Tuple[IStreamSets, str]]):
+        clients = [_get_async_client(ss) for ss
+                in set([ss for ss, _ in queries_])]
```

Comment:
I guess, you can use set comprehension directly
`clients = [_get_async_client(ss) for ss in {ss for ss, _ in queries_}]`